### PR TITLE
New `make formatdb` target

### PR DESCRIPTION
**Pull request by mithro**
Adding a new `make formatdb` target to the Makefile and needed utils. The new target will,
 * Canonicalize the database by running `./utils/sort_db.py`
 * Regenerate the [`Info.md` file](database/Info.md)

### CODE REVIEW

File: utils/sort_db.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    """Sort the bit section of a segbit line.
+
+    >>> segbit_line_sort_bits("A !28_35 !27_39 27_37")
+    'A 27_37 !27_39 !28_35'
+
+    >>> segbit_line_sort_bits("B !28_35 !27_39 !27_37")
+    'B !27_37 !27_39 !28_35'
+
+    >>> segbit_line_sort_bits("C 28_35 00_00 !27_37")
+    'C 00_00 !27_37 28_35'
```
Comment (litghost) [inline utils/sort_db.py]: No, put negative bits first or last.  Do not intermix.

Comment (mithro) [reply utils/sort_db.py]: See reason in docstring of whole module.

Comment (litghost) [reply utils/sort_db.py]: But 1 hot encoding are relatively rare.  The bits that are set are more important than the bits that are cleared.

Comment (mithro) [reply utils/sort_db.py]: It also makes it hard to see when a diff has a bit flip.

Comment (mithro) [reply utils/sort_db.py]: Also, when you sort this way, a number of physical structures become much clearer (specially around chained muxes).

**Pull request merged**